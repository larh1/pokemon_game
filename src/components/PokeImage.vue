<template>
<h2 class="fw-bold">¿Quién es ese Pokemon?</h2>
<div class="img-container">
    <img v-if="isVisible" class="img-show fade-in" :src="imgSrc" alt="Pokemon" height="200">
    <img v-else class="img-hint" :src="imgSrc" alt="Pokemon" height="200">
</div>
</template>

<script>
export default
{
    props:
    {
        pokeId:
        {
            type: Number,
            required: true,
        },
        isVisible:
        {
            type: Boolean,
            required: true,
            default: false,
        }
    },
    data()
    {
        return {}
    },
    computed:
    {
        imgSrc()
        {
            return "https://raw.githubusercontent.com/PokeAPI/" +
                "sprites/master/sprites/pokemon/other/dream-world/" +
                `${this.pokeId}.svg`;
        }
    }
}
</script>

<style scoped>
.img-container {
    height: 200px;
}

.img-hint {
    filter: brightness(0);
}

img {
    left: 0;
    right: 0;
    margin-left: auto;
    margin-right: auto;
    position: absolute;
}
</style>
