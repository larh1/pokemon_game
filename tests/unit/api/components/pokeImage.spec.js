import PokeImage from "@/components/PokeImage";
import { shallowMount } from "@vue/test-utils";

describe("PokeImage",() =>
{
    test("Debe coincidir con el spanshot",() =>
    {
        // Crear wrapper
        const wrapper = shallowMount(PokeImage,{
            props:
            {
                pokeId: 1,
                isVisible: false
            }
        });
        expect(wrapper.html()).toMatchSnapshot();
    });

    test("Debe mostrar la imagen oculta y el pokemon 20",() =>
    {
        // Crear wrapper
        const wrapper = shallowMount(PokeImage,{
            props:
            {
                pokeId: 20,
                isVisible: true,
            }
        });

        // Buscar las 2 imagenes
        const [img1,img2] = wrapper.findAll("img");
        expect(img1).toBeTruthy(); // Se muestra 1
        expect(img2).toBe(undefined); // Se oculta la otra
        // La img 1 es visible
        expect(img1.classes()).toContain("img-show");
    });

    test("Debe mostrar la imagen si isVisible=true",() =>
    {
        // Crear wrapper
        const wrapper = shallowMount(PokeImage,{
            props:
            {
                pokeId: 20,
                isVisible: true,
            }
        });

        // Buscar las 2 imagenes
        const img1 = wrapper.find("img");
        expect(img1).toBeTruthy(); // Se muestra 1
        // La img 1 es visible
        expect(img1.classes()).toContain("img-show");
    });

    test("El source de la imÃ¡gen debe ser correcto",() =>
    {
        // Crear wrapper
        const pokeId = 20;
        const wrapper = shallowMount(PokeImage,{
            props:
            {
                pokeId,
                isVisible: true,
            }
        });

        // Buscar las 2 imagenes
        const img1 = wrapper.find("img");
        const src = "https://raw.githubusercontent.com/PokeAPI/" +
            "sprites/master/sprites/pokemon/other/dream-world/" +
            `${ pokeId }.svg`;
        expect(img1.attributes("src")).toBe(src);
    });
});