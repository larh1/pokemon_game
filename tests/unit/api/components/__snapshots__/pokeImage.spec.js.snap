// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokeImage Debe coincidir con el spanshot 1`] = `
<h2 class="fw-bold">¿Quién es ese Pokemon?</h2>
<div class="img-container"><img class="img-hint" src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/1.svg" alt="Pokemon" height="200"></div>
`;
